components/Group/controllers/GroupController.ts

```diff
@@ -0,0 +1,84 @@
+import { GroupService } from '../services/GroupService';
+import { Request, Response } from 'express';
+import { UserService } from '../../User/services/UserService';
+
+
+export default class GroupController {
+	public static async getAllGroups(req: Request, res: Response) {
+		try {
+			const groups = await GroupService.getAllGroups();
+			return res.status(200).json(groups);
+		} catch(err) {
+			res.status(500).json({ error: err.message });
```

Comment:
Regarding error messages. Let's imagine that our server during this request will generate some error with sensitive data. And we will show it to the user. Not secure because we cannot control it.
Of course, you need this message in your logs for further investigation, but don't pass it to the user. Simply answer with 500 and a message, like 'Unable to get groups. Server error'